### ISSUE TITLE
quoting bug on variable assignment in while

### ISSUE THREAD

**Issue by derekkeoughan-afk** [koalaman/shellcheck#3296]
#### For bugs
- Rule Id (if any, e.g. SC1000): 
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash

find "./" -type f -iname "*.dd" | while read -r "name1" ; do echo "$name1"; \mv --backup=t -v -- "$name1" /mnt/drive1/data/zz-rec-dd/ ; done
```

#### Here's what shellcheck currently says:
```
No issues detected!

```
#### Here's what I wanted or expected to see:

...this variable name1 is never assigned because it's in quotes... as shellcheck suggested via SC2086 for the remainder of them - if this first one is in quotes by mistake, it's not flagged

do I need more coffee?

TODO: Describe expected/desired output

<img width="987" height="193" alt="Image" src="https://github.com/user-attachments/assets/88a4809c-0f11-46d8-9121-af92f73e5722" />

**Comment by derekkeoughan-afk** [comment]
my usual mistake is to use it before it's assigned - but here it's suggesting it be double-quoted, too

<img width="981" height="254" alt="Image" src="https://github.com/user-attachments/assets/31a683e6-0d1b-4c4a-aa24-c910bee53f52" />

**Comment by juliyvchirkov** [comment]
> #### Here's a snippet or screenshot that shows the problem:
> #!/bin/bash
> 
> find "./" -type f -iname "*.dd" | while read -r "name1" ; do echo "$name1"; \mv --backup=t -v -- "$name1" /mnt/drive1/data/zz-rec-dd/ ; done
>
> Here's what I wanted or expected to see:
>
> ...this variable name1 is never assigned because it's in quotes... as shellcheck suggested via SC2086 for the remainder of them - if this first one is in quotes by mistake, it's not flagged

Seems shellcheck warnings to double quote the variable for ``do echo $name1`` and ``mv --backup=t -v -- $name1`` cases made you confused in some way since the construct you are claiming to be problematic

```sh
#!/bin/bash

find "./" -type f -iname "*.dd" | while read -r "name1" ; do echo "$name1"; \mv --backup=t -v -- "$name1" /mnt/drive1/data/zz-rec-dd/ ; done
```

indeed is working like a charm - a shell interpreter strips the quotes from ``name1`` at ``while read -r "name1"``, and the ``$name`` variable gets populated with a path which arrives from ``find`` command

To confirm just checked your construct with ``sh``, ``bash``, ``dash``, ``ksh`` and ``zsh``, and the whole row unquotes ``name1`` as expected

Try to insert ``set -vx`` above your pipeline to inspect the flow yourself

**Comment by derekkeoughan-afk** [comment]
sorry - just my impatience... by putting it into a pipe the instant display of the file in the start folder is delayed by... 309 seconds as it walks the tree under it - I knew I needed more coffee

it does indeed work with the quotes around the variable in the "while" statement
